app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/point/DropPinTaskMapFragment.kt

```diff
@@ -30,14 +32,26 @@ class DropPinTaskMapFragment @Inject constructor() :
   override fun onMapReady(map: MapFragment) {
     super.onMapReady(map)
 
-    // Disable pan/zoom gestures if a marker has been placed on the map.
-    taskViewModel.features.observe(this) {
-      if (it.isEmpty()) {
-        map.enableGestures()
-      } else {
-        map.disableGestures()
+    viewLifecycleOwner.lifecycleScope.launch {
+      if (taskViewModel.unifyCaptureLocationTask) {
+        taskViewModel.initLocationUpdates(getMapViewModel())
       }
     }
+
+    // Disable pan/zoom gestures if a marker has been placed on the map.
+    taskViewModel.features.observe(this) { updateGestures() }
+
+    taskViewModel.captureLocation.observe(this) { updateGestures() }
+  }
+
+  private fun updateGestures() {
+    if (
+      !taskViewModel.features.value.isNullOrEmpty() || taskViewModel.captureLocation.value == true
```

Comment:
This logic or true or false should come from the previously mentioned stream and not outside.